pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/assignment/segment/strategy/BalancedNumSegmentAssignmentStrategy.java

```diff
@@ -51,6 +52,10 @@ public void init(HelixManager helixManager, TableConfig tableConfig) {
     SegmentsValidationAndRetentionConfig validationAndRetentionConfig = tableConfig.getValidationConfig();
     Preconditions.checkState(validationAndRetentionConfig != null, "Validation Config is null");
     _replication = validationAndRetentionConfig.getReplicationNumber();
+    // Number of replicas per partition of low-level consumers check is for the real time tables only
+    if (tableConfig.getTableType() == TableType.REALTIME) {
```

Comment:
Let's add the `TODO` comment that we can clean this up once the table config has the new API that picks up the correct replication depending on the table type.

Reply:
added a todo and a unit test.